src/coreclr/vm/jithelpers.cpp

```diff
@@ -5445,12 +5445,16 @@ void F_CALL_CONV HCCALL3(JIT_ReversePInvokeEnterTrackTransitions, ReversePInvoke
         thread->m_fPreemptiveGCDisabled.StoreWithoutBarrier(1);
         if (g_TrapReturningThreads.LoadWithoutBarrier() != 0)
         {
-            JIT_ReversePInvokeEnterRare2(frame, _ReturnAddress());
+            // If we're in an IL stub, we want to trace the address of the target method,
+            // not the next instruction in the stub.
+            JIT_ReversePInvokeEnterRare2(frame, pMD->IsILStub() && secretArg != NULL ? (void*)frame->pMD->GetMultiCallableAddrOfCode() : _ReturnAddress());
```

Comment:
It is dangerous to call `MethodDesc::GetMultiCallableAddrOfCode`. This method can take locks and throw exceptions.

Reply:
I've updated this to use the same code-path that the old system used.